zerver/lib/thumbnail.py

```diff
@@ -324,3 +319,90 @@ def split_thumbnail_path(file_path: str) -> tuple[str, BaseThumbnailFormat]:
     assert thumbnail_format is not None
     path_id = "/".join(path_parts[1:])
     return path_id, thumbnail_format
+
+
+def get_user_upload_previews(realm_id: int, content: str) -> dict[str, tuple[str, bool] | None]:
+    matches = re.findall(r"/user_uploads/(\d+/[/\w.-]+)", content)
```

Comment:
Is there anything we need to be careful with in this check around the possibility that the `path_id` has been HTML/URL-encoded differently in the message body from the `path_id` we have in the database?

Reply:
I'll check URI-escaping on this -- the last segment of the path_id can certainly contain pretty arbitrary characters.

Reply:
We call `saanitize_name`, which means that we strip out anything not URI safe.